python/cudf/dataframe/numerical.py

```diff
@@ -245,14 +245,13 @@ def sum(self, dtype=None):
     def product(self, dtype=None):
         return cpp_reduce.apply_reduce('product', self, dtype=dtype)
 
-    def mean(self, dtype=None):
-        return np.float64(self.sum(dtype=dtype)) / self.valid_count
-
-    def mean_var(self, ddof=1, dtype=None):
-        x = self.astype('f8')
-        mu = x.mean(dtype=dtype)
-        n = x.valid_count
-        asum = x.sum_of_squares(dtype=dtype)
+    def mean(self, dtype=np.float64):
+        return self.sum(dtype=dtype) / self.valid_count
```

Comment:
Instead of computing the sum as a float64, wouldn't we do better to compute it normally and then typecast to float64 afterwards?

Reply:
Summing as float64 would be better since integer may cause overflow.
Btw, I add float64 type cast after sum() and sum_of_squares() in a new commit because it is required if the input dtype is integer type.